backend/codesirius/tests/test_aws_client.py

```
@@ -0,0 +1,101 @@
+        """Test that different services create different instances."""
+        # Create instances for different services
+        s3_client = AWSClient("s3")
+        dynamodb_client = AWSClient("dynamodb")
+        
+        # Instances should be different
+        self.assertIsNot(s3_client, dynamodb_client)
+        # boto3.client should be called twice with different service names
+        self.assertEqual(mock_boto3_client.call_count, 2)
+
+    @patch("boto3.client")
+    def test_singleton_instances_are_stored(self, mock_boto3_client):
```

Ruff (0.8.2)

Missing return type annotation for public function `test_singleton_instances_are_stored`

Add return type annotation: `None`

(ANN201)